File: SeminarskiGymFit/GymFit/GymFit.Infrastructure.Interfaces/SearchObjects/ReservationSearchObject.cs
using GymFit.Core;
using GymFit.Core.Enums;

namespace GymFit.Infrastructure.Interfaces.SearchObjects
{
    public class ReservationSearchObject : BaseSearchObject
    {
        //dodati filter za status umjesto za spol 
        public Gender? spol { get; set; }
        public ReservationStatus? status { get; set; }
        public long? userId { get; set; }
        public long? trainerId { get; set; }
    }
}

File: SeminarskiGymFit/UI/gymfit_mobile/gymfit_mobile/lib/models/searchObjects/reservation_search_object.dart
import 'package:gymfit_mobile/models/searchObjects/base_search_object.dart';

class ReservationSearchObject extends BaseSearchObject {
  int? userId;
  int? spol;
  int? trainerId;
  int? status;
  int? PageNumber;
  int? PageSize;

  ReservationSearchObject(
      {this.spol,this.status, this.userId, this.PageNumber, this.PageSize, this.trainerId});

  ReservationSearchObject.fromJson(Map<String, dynamic> json) {
    spol = json['spol'];
    status= json['status'];
    userId = json['userId'];
    trainerId = json['trainerId'];
    PageNumber = json['PageNumber'];
    PageSize = json['PageSize'];
  }

  Map<String, dynamic> toJson() {
    final data = <String, dynamic>{};
    data['spol'] = spol;
    data['status'] = status;
    data['userId'] = userId;
    data['trainerId'] = trainerId;
    data['PageNumber'] = PageNumber;
    data['PageSize'] = PageSize;
    return data;
  }
}
